tests/jballerina-unit-test/src/test/resources/test-src/services/service_decl.bal

```diff
@@ -95,7 +117,8 @@ function getResourceAnnot(service object {} obj, string methodName, string[] pat
 } external;
 
 
-listener Listener lsn = new Listener();
+type LE Listener|EmptyListener|error;
+listener LE lsn = new Listener();
```

Comment:
Shouldn't `LE` be a subtype of the `Listener` object?

The spec says
> A module listener declares a variable in a similar way to a final variable declaration, but the type of the variable is always a subtype of the Listener object type, and it has the additional semantic of registering the variable's value with the module as a listener.

Reply:
I don't this the spec is updated with this https://github.com/ballerina-platform/ballerina-spec/issues/671

Reply:
What's mentioned in the issue is re: the expression, right?

> We could make listener-decl do the same thing: allow the expression in a listener to be of type L|E, where L is a listener and E is an error; if the expression results in an error, then module initialization fails.

My understanding was that a listener init can now return error;

```ballerina
class Listener {
    function init() returns error? {

    }
    
    ...
}
```

and if it is used in a listener-decl, and `new` actually returns `error`, module init fails?

But the listener declaration would still have just the listener as the type.

```ballerina
listener Listener ln = new;
```

If we allow the listener's type to include error, module initialization doesn't have to fail, right?

Reply:
Yes you are correct